### ISSUE TITLE
Rename argument dims of torch.flip to dim

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#18095]
Functions like `torch.sum` and `torch.mean` support multi-dim through argument `dim` accepting a tuple or a list.

`torch.flip` instead asks for an argument named `dims`. Yes, it doesn't do reduction, but still it processes multiple dimensions, for the sake of consistency, I propose it supports `dim` as well (and maybe `dims` may be deprecated).

https://pytorch.org/docs/master/torch.html#torch.flip says that `dims (a list or tuple) – axis to flip on` is a list or tuple, but in fact a scalar works as well, so explicit `x.flip(dim = 1)` does make sense.

**Comment by vishwakftw** [comment]
FTR, this actually works for the `flip` method, but not the function

```python
import torch

a = torch.randn(3, 3)
a.flip(0)  # works fine
a.flip([0, 1])  # also works fine
torch.flip(a, 0)  # doesn't work
```

**Comment by t-vi** [comment]
Personally, I'd think we should standardize on dim and probably just break things to rename the argument...

**Comment by ezyang** [comment]
This sounds fine. Some other evidence in favor of dim: Numpy names it axis (and it's not pluralized).

If this isn't urgent, @ifedan may look into it. I will happily review a PR submitted to do this.

**Comment by vadimkantorov** [comment]
Also would be cool to support `x[::-1]` via `torch.flip` meanwhile negative strides are not supported (also needs discussion, since it does copy, while in NumPy it's copy-free)

**Comment by fmassa** [comment]
I don't think we should alias negative striding with `torch.flip`, precisely because it doesn't share storage, and might be surprising for used coming from numpy

**Comment by vadimkantorov** [comment]
Then maybe the error message for neg strides should hint the user to look at `torch.flip`

**Comment by ezyang** [comment]
@ailzhang says: "it surprised me that a.flip(0) works fine but torch.flip(a, 0) doesn't. And this dim related args are pretty common in our user facing APIs, I feel it'd super useful if we can have a story like what and how dims args are supported in method/functions, as in general make it consistent across other APIs. "

**Comment by vadimkantorov** [comment]
@ezyang @soumith The link from description stopped working:

Works: https://pytorch.org/docs/stable/torch.html#torch.flip 
Breaks: https://pytorch.org/docs/master/torch.html#torch.flip 

I think something has changed in docs generation. It may or may not be important / intentional, just  wanted to bring this to attention

**Comment by t-vi** [comment]
They're now separate pages: https://pytorch.org/docs/master/generated/torch.flip.html#torch.flip

**Comment by vadimkantorov** [comment]
@soumith @ezyang If this new urls are decided, it would be nice to at least have server (+javascript if needed) redirects in place to not break forum links. Especially if this scheme graduates to "stable" channel

**Comment by ezyang** [comment]
@vadimkantorov Yes, this will be a release blocker for next release. Can you file a new issue, and cc @mattip to it?

**Comment by mruberry** [comment]
The divergence between method and function variants of flip suggests there's an error in codegen. This issue reproduces in PyTorch 1.7.

[NumPy's flip](https://numpy.org/doc/stable/reference/generated/numpy.flip.html) does allow the dims argument to be an integer. We should update our flip to be the same. Updating title of this issue.
